File: basic types/null/null.py
# 空值
# python语言有空值的概念, 空值为None.
array = [1, 2, 3, 4, 5]
index = [0, 2, 4, 6]

def get_element_by_index(ind):
    if ind < 0 or ind >= len(array):
        return None
    return array[ind]

sum = 0
for value in index:
    cur = get_element_by_index(value)
    if cur is not None: # 需要检查函数返回的变量是否为空，不检查也可以编译，但程序有崩溃的危险
        sum += cur
print("sum: ", sum)

File: basic types/null/null.swift
// 空值
// swift语言没有空值的概念,只有可空类型。
// nil  => 不存在值
// swift将可空类型直接放入了基础语法中，用起来更加自然

let array = [1, 2, 3, 4, 5]
let index = [0, 2, 4, 6]

// 返回可空类型
func get_element_by_index(_ ind: Int) -> Int? {
    if ind < 0 || ind >= array.count {
        return nil
    }
    return array[ind]
}

var sum = 0
for value in index {
    let cur = get_element_by_index(value)
    if let value = cur { // 需要显式检查函数返回的值，否则因为类型不同无法编译
        sum += value
    }
}
print("sum: ", sum)
